tests/commit/test_commit_spec.lua

```diff
@@ -0,0 +1,288 @@
+		assert.are.equal(Header,Header_A)
+	end)
+
+	it("Tests Commit Summary", function()
+		local Commit_A_summary = lib.luagit2_commit_summary(Commit_A)
+		assert.are.equal("Added File",Commit_A_summary)
+	end)
+
+	it("Tests Commit Tree", function()
+		local Commit_A_tree = lib.luagit2_commit_tree(Commit_A)
+		local tree_id = lib.luagit2_tree_id(Commit_A_tree)
+
+		assert.are.equal(current_write_tree_id, lib.luagit2_oid_tostr_s(tree_id))
+		lib.luagit2_tree_free(Commit_A_tree)
+	end)
+
+	it("Tests Commit Tree Id",function()
+		-- Please refer libgit2's original documentation for this
+		local tree_id = lib.luagit2_tree_id(Commit_A)
+		assert.are.equal(Commit_Id_A_string, lib.luagit2_oid_tostr_s(tree_id))
+	end)
+
+	it("Tests Creating a Commmit and update head ",function ()
+		-- Get parent from most recent commit
+		local parent_tree_id = lib.luagit2_oid_fromstr("1b49a09ae5e707b8ed48e6048aa1901138838944")
+		local parent_tree = lib.luagit2_tree_lookup(repo,parent_tree_id)
+
+		-- Create a new Commit and get commit Id
+		local new_commit_id =  lib.luagit2_commit_create_update_head(repo,Author_Sign_A,
+			Commmitter_Sign_A,"New Commit From luagit2",parent_tree,Commit_A)
+
+		print("\n New Commit Id : " .. lib.luagit2_oid_tostr_s(new_commit_id) .. "\n")
+
+		local new_commit_string = lib.luagit2_oid_tostr_s(new_commit_id)
+		
+		-- Try to lookup that Id in repo
+		-- A test of whether the commit was actually created or not
+		local new_commit = lib.luagit2_commit_lookup(repo,new_commit_id)
+		-- Check the lookedup Commit is the one that we created
+		local lookedup_commit_id = lib.luagit2_commit_id(new_commit)
+
+		assert.are.equal(new_commit_string,lib.luagit2_oid_tostr_s(lookedup_commit_id))
+
+	end)
+
+	it("Tests Creating a Commmit and update none ",function ()
+		-- Get parent from most recent commit
+		local parent_tree_id = lib.luagit2_oid_fromstr("1b49a09ae5e707b8ed48e6048aa1901138838944")
+		local parent_tree = lib.luagit2_tree_lookup(repo,parent_tree_id)
+
+		-- Create a new Commit and get commit Id
+		local new_commit_id =  lib.luagit2_commit_create_update_none(repo,Author_Sign_A,
+			Commmitter_Sign_A,"New Commit From luagit2",parent_tree,Commit_A)
+
+		print("\n New Commit Id : " .. lib.luagit2_oid_tostr_s(new_commit_id) .. "\n")
+
+		local new_commit_string = lib.luagit2_oid_tostr_s(new_commit_id)
+		
+		-- Try to lookup that Id in repo
+		-- A test of whether the commit was actually created or not
+		local new_commit = lib.luagit2_commit_lookup(repo,new_commit_id)
+		-- Check the lookedup Commit is the one that we created
+		local lookedup_commit_id = lib.luagit2_commit_id(new_commit)
+
+		assert.are.equal(new_commit_string,lib.luagit2_oid_tostr_s(lookedup_commit_id))
+
+	end)
+
+	pending(" Tests extract commit signature ", function()
+		local sign_buf = lib.luagit2_commit_extract_signature(repo,Commit_Id_A)
+		print(lib.luagit2_buf_details(sign_buf))
+	end)
+
+	pending("Tests commit time",function()
```

Comment:
@Etiene mam, this too need to be worked upon. I will rewrite this particular function. so that a string equivalent of time value is returned.

Reply:
ok! :)